apps/veil/src/pages/tournament/ui/delegator-table/delegator-rewards.tsx

```diff
@@ -21,39 +51,134 @@ export const DelegatorRewards = () => {
-              LPs Rewards
-            </SegmentedControl.Item>
-            <SegmentedControl.Item style='filled' value='voting'>
-              Voting Rewards
-            </SegmentedControl.Item>
-          </SegmentedControl>
-        </div>
-
-        {tab === 'lp' && <LpRewards />}
+export const DelegatorRewards = ({ address }: { address: Address }) => {
+  const getMetadata = useGetMetadata();
+  const [page, setPage] = useState(BASE_PAGE);
+  const [limit, setLimit] = useState(BASE_LIMIT);
+  const { getTableHeader, sortBy } = useSortableTableHeaders<SortKey>('epoch', 'desc');
 
-        {tab === 'voting' && <VotingRewards />}
+  const onLimitChange = (l: number) => {
+    setLimit(l);
+    setPage(BASE_PAGE);
+  };
+  const { data, isPending, error } = useSpecificDelegatorRewards({
+    address,
+    page,
+    limit,
+    sortDirection: sortBy.direction,
+    sortKey: sortBy.key !== '' ? sortBy.key : undefined,
+  });
+  if (error) {
+    throw error;
+  }
+  if (isPending) {
+    const totalChild = (
+      <div className='w-24 h-10'>
+        <Skeleton />
       </div>
+    );
+    const tableChild = (
+      <TableLayout getTableHeader={getTableHeader}>
+        {new Array(BASE_LIMIT).fill({}).map((_, i) => {
+          return (
+            <div key={`loading-${i}`} className='grid grid-cols-subgrid col-span-4'>
+              <TableCell cell loading={true}>
+                undefined
+              </TableCell>
+              <TableCell cell loading={true}>
+                undefined
+              </TableCell>
+              <TableCell cell loading={true}>
+                undefined
+              </TableCell>
+              <TableCell cell loading={true}>
+                <Density slim>
+                  <Button iconOnly icon={ChevronRight} disabled={true}>
+                    Go to voting reward information
+                  </Button>
+                </Density>
+              </TableCell>
+            </div>
+          );
+        })}
+      </TableLayout>
+    );
+    return <Layout totalChild={totalChild} tableChild={tableChild} />;
+  }
+
+  const total = toValueView({ value: data.total, getMetadata });
+  const totalChild = (
+    <div className='flex items-center gap-4 [&_span]:font-mono [&_span]:text-3xl'>
+      <Density sparse>
+        <ValueViewComponent valueView={total} priority='tertiary' />
+      </Density>
     </div>
   );
+  const tableChild = (
+    <>
+      <TableLayout getTableHeader={getTableHeader}>
+        {data.rewards.map(({ epoch, value, vote }) => {
+          const rowKey = `epoch-${epoch}`;
+
+          const metadata = getMetadata(vote.asset);
+          const valueView = toValueView(
+            metadata
+              ? { metadata, amount: new Amount({}) }
+              : { assetId: vote.asset, amount: new Amount({}) },
+          );
+
+          return (
+            <div key={rowKey} className='grid grid-cols-subgrid col-span-4'>
```

Comment:
todo: the whole row should be a link to the `/tournament/${epoch}` page with a highlighted background on hover